flink-clients/src/test/java/org/apache/flink/client/program/ClusterClientTest.java

```diff
@@ -134,6 +140,25 @@ public void testClusterClientSavepoint() throws Exception {
 		}
 	}
 
+	@Test
+	public void testClusterClientList() throws Exception {
+		Configuration config = new Configuration();
+		config.setString(JobManagerOptions.ADDRESS, "localhost");
+
+		TestListActorGateway gateway = new TestListActorGateway();
+		ClusterClient clusterClient = new TestClusterClient(config, gateway);
+		try {
+			CompletableFuture<Collection<JobDetails>> jobDetailsFuture = clusterClient.listJobs();
+			Collection<JobDetails> jobDetails = jobDetailsFuture.get();
+			Assert.assertTrue(gateway.messageArrived);
+			// finished jobs should be ignored
+			Assert.assertEquals(1, jobDetails.size());
```

Comment:
Shouldn't we also retrieve the finished jobs here?

Reply:
No, the LIST command only lists running/scheduled jobs.